### PR TITLE
upgrade to gulp 4 and update others dev dependencies

### PR DESCRIPTION
**Pull request by Symro**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

Hi I saw your tweet and the issue #2150. I did the migration to Gulp 4. You can test by yourself to check if it's the intended behaviour. It should I did not change that much.

What I did exactly:

- Updated some dev dependencies used in gulp
- Changed the babel node modules to be compatible with Gulp V4
- Updated the babel config inside the package.json
- Moved Autoprefixer browserlist config from gulpfile to the package.json

My node version: v10.15.4 and npm: 6.9.0. I don't know if the projects now works for "node": ">=8" and "npm": ">=3". Maybe we need to update the engine config aswell.

Cheers ! 🍻

### CODE REVIEW

**Comment by roblarsen** [comment]
That's great, thanks! I'll test this out.

File: gulpfile.babel.js [hunk 1]
```diff
@@ -120,75 +103,81 @@ gulp.task('copy:license', () =>
...
     `!${dirs.src}/css/main.css`,
     `!${dirs.src}/index.html`
-
   ], {
-
-    // Include hidden files by default
-    dot: true
-
-  }).pipe(gulp.dest(dirs.dist))
+      // Include hidden files by default
```
Comment (coliff) [inline gulpfile.babel.js]: minor nitpick here - but the indentation is a little bit off. should be 4 spaces (not 6) on lines 124 and 125 and indent of 2 spaces on line 126.

**Comment by coliff** [comment]
great PR! Nice work @Symro. I've tested this on Windows 10 Pro and no issues. Builds fine with no errors and `npm run test` completes all 41 tests successfully.

File: gulpfile.babel.js [hunk 2]
```diff
@@ -120,75 +103,81 @@ gulp.task('copy:license', () =>
...
-    ['clean', 'lint:js'],
-    'copy', 'modernizr',
-    done);
-});
+    'archive:zip'
+  )
+);
 
-gulp.task('default', ['build']);
+gulp.task('default', gulp.series('build'))
```
Comment (coliff) [inline gulpfile.babel.js]: eslint complains that this should have a semicolon at the end here.

File: gulpfile.babel.js (hunk 1 above)
Comment (Symro) [reply gulpfile.babel.js]: fixed

File: gulpfile.babel.js (hunk 2 above)
Comment (Symro) [reply gulpfile.babel.js]: fixed

**Comment by roblarsen** [comment]
This is really great. I tested with and it build with no problem. 

  * 12.9.1 
  * 10.16.3
  *  8.9.4

Thanks for coming through so quickly on this. I am really happy we were able to get this fixed.

**Comment by Symro** [comment]
> This is really great. I tested with and it build with no problem.
> 
> * 12.9.1
> * 10.16.3
> * 8.9.4
> 
> Thanks for coming through so quickly on this. I am really happy we were able to get this fixed.

You're welcome ;), I dit this kind of work a couple of months ago so it was really fun to do it again for an open source project 👍.

**Pull request merged**